apps/staking/hooks/useStakes.tsx

```diff
@@ -142,25 +148,51 @@ export function useStakes(overrideAddress?: Address) {
     const networkTime =
       net && 'block_timestamp' in net && net.block_timestamp ? data.network.block_timestamp : 0;
 
-    const stakeBlsKeys = new Set(stakesArr.map(({ pubkey_bls }) => pubkey_bls));
+    const addedBlsKeysSet = new Set(
+      stakesArr
+        .filter((stake) => parseStakeState(stake, blockHeight) !== STAKE_STATE.EXITED)
+        .map(({ pubkey_bls }) => pubkey_bls)
+    );
 
     const limitAgeJoiningTimestamp = Date.now() - CONTRIBUTION_CONTRACT.MAX_AGE_JOINING_MS;
-    const filteredContracts = contractsArr
-      .filter(({ pubkey_bls }) => !stakeBlsKeys.has(pubkey_bls.slice(2)))
-      .filter(
-        ({ node_add_timestamp }) =>
-          !node_add_timestamp || node_add_timestamp * 1000 > limitAgeJoiningTimestamp
-      );
+
+    const addedAwaitingOperatorContracts = new Set();
+
+    const filteredContracts = contractsArr.filter(({ pubkey_bls, node_add_timestamp, status }) => {
+      switch (status) {
+        case CONTRIBUTION_CONTRACT_STATUS.Finalized:
+          // Should only include the finalised contracts that have no bls key added time or have an added time recently
+          return !node_add_timestamp || node_add_timestamp * 1000 > limitAgeJoiningTimestamp;
```

Comment:
This timestamp is in seconds right? Might be good to add a docstring at it's source or rename it to have seconds in the name. I see concerting between s and ms being forgotten at somepoint (I have done this often myself)

Reply:
so `node_add_timestamp` is in seconds, it is an external value, all external timestamps are in seconds, (from backend or from smart contracts), this is the first point is used, where it is then converted to seconds in-line. not really anywhere else to do it